### PR TITLE
Option to save highscore has been provided

### PR DESCRIPTION
**Pull request by vinaykrishna64**
Now score will displayed during playing, pause and quit
Option to save highscore has been provided

### CODE REVIEW

**Comment by jajodiaraghav** [comment]
Can you please add `__pycache__` to `.gitignore`

File: main.py [hunk 1]
```diff
@@ -48,6 +55,8 @@ def pause():
                     quit()
         message("Paused", (0, 0, 0))
         message("C to continue, Q to Quit", (200, 0, 0), 40)
+                #display score on pause
```
Comment (jajodiaraghav) [inline main.py]: Fix the indent

File: main.py [hunk 2]
```diff
@@ -111,12 +120,14 @@ def get_blocks(food_rect, n):
...
     global k
+    global highscore
+    global namehighscore
     pyExit = False
     pyOver = False
 
     score = 0
     world_num = 0
-
+    scorestr = "Score:0"
```
Comment (jajodiaraghav) [inline main.py:130]: I think there should be a space after `:`

File: main.py [hunk 3]
```diff
@@ -130,8 +141,53 @@ def gameLoop():
...
+                scorewindow.title("congratulations")
+                
+                
+                Label(frame, text='you\'ve made highscore!!!!').pack(side='top')
+                v = StringVar()
+                v.set("type your name")
+                textbox = Entry(frame, textvariable = v)
+                textbox.pack(side='top')
+                
+                butok=Button(frame,text="ok",fg="black",bg="white", command = namewrite)
```
Comment (jajodiaraghav) [inline main.py]: Learn to write proper camelcased variable names. Just building a feature isn't enough. Always remember that there will be many devs working on the same project

File: main.py [hunk 4]
```diff
@@ -224,7 +280,7 @@ def gameLoop():
                 while food_collides_block(food.get_rect(), blocks):
                     food.generate_food(width, height)
 
-                print score, food.x, food.y
+                #print score, food.x, food.y
```
Comment (jajodiaraghav) [inline main.py]: Remove the line if not in use. Do not just comment them out

File: main.py [hunk 5]
```diff
@@ -236,6 +292,12 @@ def gameLoop():
         # draw the blocks.
         for block in blocks:
             block.draw(game_display, (255, 0, 0))
+        #count and display score on screen
+        tot = total(score,world_num)
```
Comment (jajodiaraghav) [inline main.py]: `tot` doesn't sound like a proper name to a variable. Please fix

**Review by jajodiaraghav — CHANGES_REQUESTED** [review]
Nice work. Left a few comments.

**Comment by vinaykrishna64** [comment]
I have committed the requested changes

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+__pycache__
```
Comment (jajodiaraghav) [inline .gitignore]: The pycache files are still present in the PR. you will have to do a git reset --soft followed by a git commit to remove them. Please always have a look at the diff before pushing and https://github.com/jajodiaraghav/Snake/pull/29/files after pushing.

**Comment by jajodiaraghav** [comment]
@vinaykrishna64 Please try to understand that I cannot merge until the `pycache` files are removed from the PR

**Comment by jajodiaraghav** [comment]
Check: https://github.com/jajodiaraghav/Snake/pull/29/files

**Comment by vinaykrishna64** [comment]
@jajodiaraghav  I did  remove the requested files

image

**Comment by jajodiaraghav** [comment]
Thanks @vinaykrishna64

**Pull request merged**